libraries/resource.rb

```diff
@@ -1,6 +1,10 @@
 module Nginx
   module Cookbook
     module ResourceHelpers
+      def init_list_resource
```

Comment:
Happy to merge as is, but I this section could do with a bit more documentation, as the why/how isn't super clear here 👍🏼

Reply:
No problem I can definitely get some comments in there, especially as it's the same pattern I've used in multiple cookbooks. 

It's just a way to allow ensuring that an inclusion list resource is always created no matter where it gets called from. But to remove the same problem as the multiple service actions only do it once to ensure you aren't overwritting yourself.
 
In hindsight it can come out of the run context and become a declare_resource due to the other method checking if it's been defined beforehand. 

I'll sort a patch version out when I've got chance.

Reply:
Actually I'll modify this, thought it'd been merged already for some reason.